### PR TITLE
chapter_11

### PR DESCRIPTION
**Pull request by azimfaiz**

### CODE REVIEW

**Comment by azimfaiz** [comment]
ready

File: activation_digest:string [hunk 1]
```diff
@@ -0,0 +1,2 @@
+      invoke  active_record
```
Comment (ngocvt-0484) [inline activation_digest:string]: delete it

File: app/assets/stylesheets/account_activations.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/account_activations.scss]: delete it

File: app/models/user.rb [hunk 3]
```diff
@@ -43,9 +46,22 @@ def forgets
...
+    
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
+  end
+  
   private
   
   def downcase_email
-    email.downcase!
+    self.email = email.downcase!
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
    self.email.downcase!
```

File: app/helpers/account_activations_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (ngocvt-0484) [inline app/helpers/account_activations_helper.rb]: remove this file because you not use

**Comment by azimfaiz** [comment]
ready

File: db/migrate/20200922074158_add_activation_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+ 
+class AddActivationToUsers < ActiveRecord::Migration[6.0]
```
Comment (saobv-2223) [inline db/migrate/20200922074158_add_activation_to_users.rb]:
```suggestion
 class AddActivationToUsers < ActiveRecord::Migration[6.0]
```

File: config/environments/development.rb [hunk 6]
```diff
@@ -31,11 +31,19 @@
...
-  # Don't care if the mailer can't send.
-  config.action_mailer.raise_delivery_errors = false
-
-  config.action_mailer.perform_caching = false
-
+  # Don"t care if the mailer can"t send.
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.delivery_method = :smtp
+  host = "localhost"
+  config.action_mailer.default_url_options = { host: host, port: '3000', protocol: 'http' }
```
Comment (saobv-2223) [inline config/environments/development.rb]: using "

**Comment by azimfaiz** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**